ABObjectCore.js

```diff
@@ -897,10 +910,20 @@ module.exports = class ABObjectCore extends ABMLClass {
       if (!labelData && this.fields().length > 0) {
          var defaultField = this.fields((f) => f.fieldUseAsLabel())[0];
          if (defaultField) labelData = "{" + defaultField.id + "}";
-         else
-            labelData = `${this.AB.rules.isUUID(rowData.id) ? "ID: " : ""}${
-               rowData.id
-            }`; // show id of row
+         else {
+            // if label is empty, then show .id
+            if (!labelData.trim()) {
+               let labelSettings = this.labelSettings || {};
+               if (labelSettings && labelSettings.isNoLabelDisplay) {
+                  labelData = L("[No Item Reported]");
```

Comment:
I'm not sure about having this as default. It seems like it might be confusing if the object is not about items. I think it needs to be more generic. Maybe "No Data"?